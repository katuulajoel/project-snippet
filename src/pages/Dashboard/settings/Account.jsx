/* eslint-disable react/no-unescaped-entities */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import { Row, Col, FormGroup, Label } from "reactstrap";
import styled from "styled-components";

/* ------------------------- Component dependencies ------------------------- */
import CustomInputGroup from "../../../components/CustomInputGroup";
import Button from "../../../components/Button";
import Input from "../../../components/Input";

const Account = () => {
  return (
    <ContentSection className="account-settings">
      <form>
        <div className="header">Add admin email address for invoices</div>
        <Label>
          Email Address
          <span className="label-style">*</span>
        </Label>
        <Row>
          <Col md="12">
            <FormGroup>
              <Input value="" required />
            </FormGroup>
          </Col>
          <Col>
            <Button className="save" type="submit">
              Save
            </Button>
          </Col>
        </Row>
      </form>
      <form className="update-password-form">
        <div className="header">Change Password</div>

        <Row>
          <Col>
            <FormGroup>
              <Label>
                Current Password
                <span className="label-style">*</span>
              </Label>
              <CustomInputGroup
                className="password2"
                variant="password2"
                placeholder="Enter your password"
                value=""
                required
              />
            </FormGroup>
          </Col>

          <Col>
            <FormGroup>
              <Label>
                New Password
                <span className="label-style">*</span>
              </Label>
              <CustomInputGroup
                className="password2"
                variant="password2"
                placeholder="Enter new password"
                value=""
                required
              />
            </FormGroup>
          </Col>
        </Row>
        <Row>
          <Col>
            <Button className="save" type="submit">
              Save
            </Button>
          </Col>
        </Row>
      </form>
      <form>
        <div className="header">Change Email Address</div>
        <Row>
          <Col md="6">
            <Label>
              New Email Address
              <span className="label-style">*</span>
            </Label>
            <FormGroup>
              <Input placeholder="Enter new email address" value="" required />
            </FormGroup>
          </Col>
          <Col md="6">
            <Label>
              Password
              <span className="label-style">*</span>
            </Label>
            <FormGroup>
              <CustomInputGroup
                className="password2"
                variant="password2"
                placeholder="Enter your password"
                value=""
                required
              />
            </FormGroup>
          </Col>
        </Row>

        <Row>
          <Col>
            <Button className="save" type="submit">
              Save
            </Button>
          </Col>
        </Row>
      </form>
      <form>
        <div className="header">Deactivate account</div>
        <div>
          <Button className="save" type="submit">
            {" "}
            Deactivate my account
          </Button>
        </div>
      </form>
    </ContentSection>
  );
};

const ContentSection = styled.div`
  width: 600px;
  padding-top: 70px;
  margin: auto;

  .label-style {
    color: #da3451;
    padding-left: 2px;
  }
  .header {
    font-weight: 600;
    font-size: 16px;
    line-height: 24px;
    color: #151a30;
    margin-bottom: 24px;
  }

  form {
    border-bottom: 1px solid #edf1f7;
    padding-bottom: 24px;

    &:last-of-type {
      border-bottom: none;
      padding-bottom: 0px;
    }

    .control-label {
      font-weight: 500;
      font-size: 14px;
      line-height: 21px;
      color: #151a30;
    }

    input,
    textarea,
    select {
      background: #ffffff;
      border: 1px solid rgba(194, 204, 217, 0.25);
      box-sizing: border-box;
      box-shadow: none;
      border-radius: 4px;
      margin-bottom: 16px;
      font-size: 16px;
      line-height: 24px;
      color: #3e4857;
    }

    .password2 {
      border-right: none;
      box-shadow: none;

      input {
        border-right: none;
      }

      .input-group-prepend {
        .input-group-text {
          padding: 0px;
          button {
            min-width: 45px;
            background-color: #fff;
            border: 1px solid rgba(194, 204, 217, 0.25);
            border-radius: 0px 4px 4px 0px;
            border-left: none;
            i {
              font-size: 16px;
              color: #8f9bb3;
            }
          }
        }
      }
    }

    .tag-input {
      .input-group {
        box-shadow: none;
        input {
          background: #ffffff;
          border: 1px solid rgba(194, 204, 217, 0.25);
          box-sizing: border-box;
          box-shadow: none;
          border-radius: 4px;
          margin-bottom: 16px;
          font-size: 16px;
          line-height: 24px;
          color: #3e4857;
          &::placeholder {
            padding: 8px 16px;
          }
        }
      }
    }

    .save {
      box-shadow: none;
      border: none;
      background: rgba(6, 46, 100, 0.05);
      border-radius: 4px;
      font-weight: 600;
      font-size: 16px;
      line-height: 19px;
      text-align: center;
      color: #062e64;

      &:disabled {
        color: rgba(6, 46, 100, 0.25);
      }
    }
  }
`;

export default Account;
