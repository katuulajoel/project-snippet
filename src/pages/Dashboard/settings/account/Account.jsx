/* eslint-disable react/no-unescaped-entities */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import styled from "styled-components";
import { isAdmin } from "../../../../utils/auth";
/* ------------------------- Component dependencies ------------------------- */
import AddAdminEmail from "./AddAdminEmail";
import ChangePassword from "./ChangePassword";
import ChangeEmail from "./ChangeEmail";
import Deactivate from "./Deactivate";
import {
  updateAuthUser,
  updateAccountInfo,
  updatePassword,
  deactivateAccount,
} from "../../../../redux/actions/ProfileActions";
import { openConfirm } from "../../../../utils/modals";

const Account = () => {
  const [page] = useState({
    isSaved: "",
  });
  const { user } = useSelector(({ Auth }) => Auth);
  const dispatch = useDispatch();

  const getFormData = (formdata) => {
    let data = {};
    formdata.forEach((val, key) => (data[key] = val));
    return data;
  };

  const addAdminEmail = (e) => {
    e.preventDefault();

    let formdata = new FormData(document.querySelector("#addAdminEmail"));
    let data = getFormData(formdata);
    console.log("addAdminEmail", data);

    return updateAuthUser(data)(dispatch);
  };

  const changePassword = (e) => {
    e.preventDefault();

    let formdata = new FormData(document.querySelector("#changePassword"));
    let data = getFormData(formdata);
    if (data) {
      data["new_password2"] = data["new_password1"];
    }
    console.log("changePassword", data);

    return updatePassword(data)(dispatch);
  };

  const changeEmail = (e) => {
    e.preventDefault();

    let formdata = new FormData(document.querySelector("#changeEmail"));
    let data = getFormData(formdata);
    console.log("changeEmail", data);

    return updateAccountInfo(data)(dispatch);
  };

  const deactivate = (e) => {
    e.preventDefault();

    openConfirm({
      message: (
        <div>
          <p>
            Your user account will be disabled immediately and you will be
            logged out of Tunga.
          </p>
          <p>
            Additionally, all data that's non-essential to compliance will be
            deleted from Tunga's systems within 7 days.
          </p>
        </div>
      ),
      title: "Confirm Account Deactivation",
      canClose: true,
      options: {
        ok: "Confirm",
        cancel: "Cancel",
        dismissAsCancel: true,
        size: "md",
      },
    }).then(() => deactivateAccount());
  };

  return (
    <ContentSection className="account-settings">
      {user.is_project_owner || isAdmin() ? ( // TODO: remove isAdmin()
        <AddAdminEmail onSave={addAdminEmail} />
      ) : null}

      <ChangePassword onSave={changePassword} />
      <ChangeEmail onSave={changeEmail} />
      <Deactivate page={page} deactivate={deactivate} />
    </ContentSection>
  );
};

const ContentSection = styled.div`
  width: 600px;
  padding: 40px;
  margin: auto;
  background: white;

  .label-style {
    color: #da3451;
    padding-left: 2px;
  }

  label {
    font-weight: 400;
    display: inline-block;
    margin-bottom: 0.5rem;
    font-size: 0.875rem;
  }

  .header {
    font-weight: 600;
    font-size: 16px;
    line-height: 24px;
    color: #151a30;
    margin-bottom: 24px;
  }

  form {
    border-bottom: 1px solid #edf1f7;
    padding-bottom: 24px;
    margin-bottom: 20px;

    &:last-of-type {
      border-bottom: none;
      padding-bottom: 0px;
    }

    .control-label {
      font-weight: 500;
      font-size: 14px;
      line-height: 21px;
      color: #151a30;
    }

    input,
    textarea,
    select {
      background: #ffffff;
      border: 1px solid rgba(194, 204, 217, 0.25);
      box-sizing: border-box;
      box-shadow: none;
      border-radius: 4px;
      margin-bottom: 16px;
      font-size: 16px;
      line-height: 24px;
      color: #3e4857;
    }

    .password2 {
      border-right: none;
      box-shadow: none;

      input {
        border-right: none;
      }

      .input-group-prepend {
        .input-group-text {
          padding: 0px;
          button {
            min-width: 45px;
            background-color: #fff;
            border: 1px solid rgba(194, 204, 217, 0.25);
            border-radius: 0px 4px 4px 0px;
            border-left: none;
            i {
              font-size: 16px;
              color: #8f9bb3;
            }
          }
        }
      }
    }

    .tag-input {
      .input-group {
        box-shadow: none;
        input {
          background: #ffffff;
          border: 1px solid rgba(194, 204, 217, 0.25);
          box-sizing: border-box;
          box-shadow: none;
          border-radius: 4px;
          margin-bottom: 16px;
          font-size: 16px;
          line-height: 24px;
          color: #3e4857;
          &::placeholder {
            padding: 8px 16px;
          }
        }
      }
    }

    .save {
      box-shadow: none;
      border: none;
      background: rgba(6, 46, 100, 0.05);
      border-radius: 4px;
      font-weight: 600;
      font-size: 16px;
      line-height: 19px;
      text-align: center;
      color: #062e64;
      height: 2.5rem;
      padding: 0.4375rem 1.25rem;

      &:disabled {
        color: rgba(6, 46, 100, 0.25);
      }
    }
  }
`;

export default Account;
