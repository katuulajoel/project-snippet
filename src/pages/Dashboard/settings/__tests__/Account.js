import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Account from "../Account";

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <Router>
          <Account />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
