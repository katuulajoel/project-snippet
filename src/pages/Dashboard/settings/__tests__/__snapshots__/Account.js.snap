// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Snapshot test for Login component 1`] = `
<div
  className="sc-hKFxyN hQSbBr account-settings"
>
  <form>
    <div
      className="header"
    >
      Add admin email address for invoices
    </div>
    <label
      className=""
    >
      Email Address
      <span
        className="label-style"
      >
        *
      </span>
    </label>
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <div
          className="form-group"
        >
          <div
            className="sc-gtsrHT jGBtBH"
            data-testid="input"
          >
            <input
              className="form-control"
              data-testid=""
              id=""
              label="Label"
              placeholder="input"
              required={true}
              type="input"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <button
          className="btn btn-default save  "
          form={null}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
  <form
    className="update-password-form"
  >
    <div
      className="header"
    >
      Change Password
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Current Password
            <span
              className="label-style"
            >
              *
            </span>
          </label>
          <div
            className="input-group  password2 "
          >
            <div
              className="sc-gtsrHT jGBtBH"
              data-testid="input"
            >
              <input
                className="form-control"
                data-testid=""
                id=""
                label="Label"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Enter your password"
                required={true}
                type="password"
                value=""
              />
              <button
                className="input-icon "
                onClick={[Function]}
                type="button"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 5.41667C11.5323 5.41157 13.0349 5.83899 14.3351 6.64977C15.6353 7.46056 16.6804 8.6218 17.35 10C15.975 12.8083 13.1667 14.5833 10 14.5833C6.83337 14.5833 4.02504 12.8083 2.65004 10C3.31972 8.6218 4.36473 7.46056 5.66494 6.64977C6.96515 5.83899 8.46776 5.41157 10 5.41667ZM10 3.75C5.83337 3.75 2.27504 6.34167 0.833374 10C2.27504 13.6583 5.83337 16.25 10 16.25C14.1667 16.25 17.725 13.6583 19.1667 10C17.725 6.34167 14.1667 3.75 10 3.75ZM10 7.91667C10.5526 7.91667 11.0825 8.13616 11.4732 8.52686C11.8639 8.91756 12.0834 9.44747 12.0834 10C12.0834 10.5525 11.8639 11.0824 11.4732 11.4731C11.0825 11.8638 10.5526 12.0833 10 12.0833C9.44751 12.0833 8.9176 11.8638 8.5269 11.4731C8.1362 11.0824 7.91671 10.5525 7.91671 10C7.91671 9.44747 8.1362 8.91756 8.5269 8.52686C8.9176 8.13616 9.44751 7.91667 10 7.91667ZM10 6.25C7.93337 6.25 6.25004 7.93333 6.25004 10C6.25004 12.0667 7.93337 13.75 10 13.75C12.0667 13.75 13.75 12.0667 13.75 10C13.75 7.93333 12.0667 6.25 10 6.25Z"
                    fill="#8F9BB3"
                  />
                </svg>
              </button>
            </div>
            <span
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <button
                  className="btn btn-icon sc-bdnxRM gIqEUJ  "
                  form={null}
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="tg-ic-outline-remove-red-eye tunga-ic-sz-md "
                  />
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            New Password
            <span
              className="label-style"
            >
              *
            </span>
          </label>
          <div
            className="input-group  password2 "
          >
            <div
              className="sc-gtsrHT jGBtBH"
              data-testid="input"
            >
              <input
                className="form-control"
                data-testid=""
                id=""
                label="Label"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Enter new password"
                required={true}
                type="password"
                value=""
              />
              <button
                className="input-icon "
                onClick={[Function]}
                type="button"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 5.41667C11.5323 5.41157 13.0349 5.83899 14.3351 6.64977C15.6353 7.46056 16.6804 8.6218 17.35 10C15.975 12.8083 13.1667 14.5833 10 14.5833C6.83337 14.5833 4.02504 12.8083 2.65004 10C3.31972 8.6218 4.36473 7.46056 5.66494 6.64977C6.96515 5.83899 8.46776 5.41157 10 5.41667ZM10 3.75C5.83337 3.75 2.27504 6.34167 0.833374 10C2.27504 13.6583 5.83337 16.25 10 16.25C14.1667 16.25 17.725 13.6583 19.1667 10C17.725 6.34167 14.1667 3.75 10 3.75ZM10 7.91667C10.5526 7.91667 11.0825 8.13616 11.4732 8.52686C11.8639 8.91756 12.0834 9.44747 12.0834 10C12.0834 10.5525 11.8639 11.0824 11.4732 11.4731C11.0825 11.8638 10.5526 12.0833 10 12.0833C9.44751 12.0833 8.9176 11.8638 8.5269 11.4731C8.1362 11.0824 7.91671 10.5525 7.91671 10C7.91671 9.44747 8.1362 8.91756 8.5269 8.52686C8.9176 8.13616 9.44751 7.91667 10 7.91667ZM10 6.25C7.93337 6.25 6.25004 7.93333 6.25004 10C6.25004 12.0667 7.93337 13.75 10 13.75C12.0667 13.75 13.75 12.0667 13.75 10C13.75 7.93333 12.0667 6.25 10 6.25Z"
                    fill="#8F9BB3"
                  />
                </svg>
              </button>
            </div>
            <span
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <button
                  className="btn btn-icon sc-bdnxRM gIqEUJ  "
                  form={null}
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="tg-ic-outline-remove-red-eye tunga-ic-sz-md "
                  />
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <button
          className="btn btn-default save  "
          form={null}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
  <form>
    <div
      className="header"
    >
      Change Email Address
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <label
          className=""
        >
          New Email Address
          <span
            className="label-style"
          >
            *
          </span>
        </label>
        <div
          className="form-group"
        >
          <div
            className="sc-gtsrHT jGBtBH"
            data-testid="input"
          >
            <input
              className="form-control"
              data-testid=""
              id=""
              label="Label"
              placeholder="Enter new email address"
              required={true}
              type="input"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <label
          className=""
        >
          Password
          <span
            className="label-style"
          >
            *
          </span>
        </label>
        <div
          className="form-group"
        >
          <div
            className="input-group  password2 "
          >
            <div
              className="sc-gtsrHT jGBtBH"
              data-testid="input"
            >
              <input
                className="form-control"
                data-testid=""
                id=""
                label="Label"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Enter your password"
                required={true}
                type="password"
                value=""
              />
              <button
                className="input-icon "
                onClick={[Function]}
                type="button"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 5.41667C11.5323 5.41157 13.0349 5.83899 14.3351 6.64977C15.6353 7.46056 16.6804 8.6218 17.35 10C15.975 12.8083 13.1667 14.5833 10 14.5833C6.83337 14.5833 4.02504 12.8083 2.65004 10C3.31972 8.6218 4.36473 7.46056 5.66494 6.64977C6.96515 5.83899 8.46776 5.41157 10 5.41667ZM10 3.75C5.83337 3.75 2.27504 6.34167 0.833374 10C2.27504 13.6583 5.83337 16.25 10 16.25C14.1667 16.25 17.725 13.6583 19.1667 10C17.725 6.34167 14.1667 3.75 10 3.75ZM10 7.91667C10.5526 7.91667 11.0825 8.13616 11.4732 8.52686C11.8639 8.91756 12.0834 9.44747 12.0834 10C12.0834 10.5525 11.8639 11.0824 11.4732 11.4731C11.0825 11.8638 10.5526 12.0833 10 12.0833C9.44751 12.0833 8.9176 11.8638 8.5269 11.4731C8.1362 11.0824 7.91671 10.5525 7.91671 10C7.91671 9.44747 8.1362 8.91756 8.5269 8.52686C8.9176 8.13616 9.44751 7.91667 10 7.91667ZM10 6.25C7.93337 6.25 6.25004 7.93333 6.25004 10C6.25004 12.0667 7.93337 13.75 10 13.75C12.0667 13.75 13.75 12.0667 13.75 10C13.75 7.93333 12.0667 6.25 10 6.25Z"
                    fill="#8F9BB3"
                  />
                </svg>
              </button>
            </div>
            <span
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <button
                  className="btn btn-icon sc-bdnxRM gIqEUJ  "
                  form={null}
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="tg-ic-outline-remove-red-eye tunga-ic-sz-md "
                  />
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <button
          className="btn btn-default save  "
          form={null}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
  <form>
    <div
      className="header"
    >
      Deactivate account
    </div>
    <div>
      <button
        className="btn btn-default save  "
        form={null}
        type="submit"
      >
         
        Deactivate my account
      </button>
    </div>
  </form>
</div>
`;
